### DISCUSSION TITLE
Implement crash recovery?

### DISCUSSION THREAD

**Discussion by stefonarch** [labwc/labwc#2002] [General]
https://github.com/ferdi265/wl-restart is released. Although I never had seen a crash in labwc - can this be implemented? Qt apps will survive, after the window with the nested labwc is closed we see:
```
$ labwc -C /tmp/ -s fpad
00:00:03.980 [../src/view.c:757] view has no output, not positioning
qt.qpa.wayland: Creating a fake screen in order for Qt not to crash
```

**Comment by Consolatis** [comment]
> Your compositor must support Wayland socket handover. The socket is passed to the compositor with the options --socket NAME --wayland-fd FD.

That looks doable (although I am not sure I like this syntax compared to a env var). We would need to play around a bit with how wlroots creates the socket. From memory it should be possible to supply a fd as the listening socket but I didn't verify it.

I am also not sure why the external application is required at all and why it needs the socket handover part. If Qt6 apps simply try to reconnect it should be enough to `SIGKILL` the nested labwc instance and then start it again.
